Cauldron/Assets/Scripts/SO/Recipe.cs

```diff
@@ -48,5 +48,32 @@ public override bool Localize(Language language)
 
             return false;
         }
+        
+        public bool SearchRecipe(Ingredients ingredients, Ingredients ingredients1, Ingredients ingredients2)
```

Comment:
This function can be simplified to a couple of lines if I'm not mistaken)

hashSet = new HashSet(1, 2, 3)
return hashSet.Equals(RecipeIngredients.ToHashSet())